app/api/check_cumulative.py

```diff
@@ -2,116 +2,117 @@
-
-                            if not onlyThisWeek:
-                                writer.writerow(row_to_write)
-
-                            elif onlyThisWeek and collection_date == this_week:
-                                writer.writerow(row_to_write)
-
-                    last_cumes[k] = (current_cume, collection_date)
-
-        t1 = time()
-
-        for state in states:
-
-            url = utils.csv_url_for_sheets_url(state)
-            df = pd.read_csv(url)
-
-            state_name = set(df['State']).pop()
-            flask.current_app.logger.info('Generating quality report for %s' % state_name)
-
-            utils.standardize_data(df)
-            extra_data_standardization(df, state_name)
-
-            check_cumulative(df)
-
-        t2 = time()
-
-        # this will go into the lambda logs
-        flask.current_app.logger.info('Generating quality report took %.1f seconds' % (t2 - t1))
-
-def transform_RI(cume_res_pos):
-    return int(cume_res_pos.split(' ')[0])
+def check_cumulative(df, onlyThisWeek=False):
+    fieldnames = [
+        'State',
+        'Facility',
+        'cumulative_field_decresed',
+        'Date',
+        'current_cumulative_value',
+        'Prev Date',
+        'prev_cumulative_value',
+        'CTP_Facility_Type'
+    ]
+
+    errors = pd.DataFrame(columns=fieldnames)
+
+    state_name = set(df['State']).pop()
+    flask.current_app.logger.info('Finding decreased cumulative fields for %s' % state_name)
+
+    extra_data_standardization(df, state_name)
+
+    collection_dates = df[['Date']].drop_duplicates()
+    this_week = collection_dates.max()
+
+    facilities = df[['Facility', 'County', 'City', 'CTP_Facility_Type']].drop_duplicates()
+    cume_cols = [x for x in df.columns if x.startswith('Cume_')]
+
+    for _, row in facilities.iterrows():
+        facility = df.loc[
+            (df.Facility == row.Facility) &
+            (df.County == row.County) &
+            (df.City == row.City) &
+            (df.CTP_Facility_Type == row.CTP_Facility_Type)].reset_index()
+
+        facility = facility.sort_values(by=['Date'])
+
+        for f_index, f_row in facility.iterrows():
+            if f_index == 0:
+                continue
+
+            for col in cume_cols:
+                row_to_write = None
+                if f_row[col] != -1 and f_row[col] < int(facility.loc[f_index-1, col]):
+                    row_to_write = {'State': f_row['State'],
+                        'Facility': f_row['Facility'],
+                        'Date': f_row['Date'],
+                        'cumulative_field_decresed': col,
+                        'current_cumulative_value': int(f_row[col]),
+                        'Prev Date': facility.loc[f_index-1, 'Date'],
+                        'prev_cumulative_value': int(facility.loc[f_index-1, col]),
+                        'CTP_Facility_Type': f_row['CTP_Facility_Type']}
+
+                if row_to_write:
```

Comment:
you don't need this if statement I think if you just make the conditional part of the previous code block, that creates the `row_to_write`